### DISCUSSION TITLE
Adding links to modules mentioned in sources

### DISCUSSION THREAD

**Discussion by BrianMusson** [gruntwork-io/knowledge-base#342] [Feedback]
Hello, I find it difficult to navigate to the modules mentioned in the `source =` and `base_source_url =` declarations. Could we get the URLs added just above the source line? I think this would be easy to do an save a lot of clicking around when troubleshooting. It would be handy to have it for the `live/`, as well as the `modules/` sections of the service catalog. I am using the master here in the example, but if its being generated then it could reference the tagged version from the source line.

Something like this:

```
# ---------------------------------------------------------------------------------------------------------------------
# CREATE S3 BUCKETS FOR ALB LOGS
# When Ingress is enabled (expose_type = internal or external), we need to create S3 buckets to store the ALB access
# logs.
# ---------------------------------------------------------------------------------------------------------------------

module "alb_access_logs_bucket" {
  # https://github.com/gruntwork-io/terraform-aws-monitoring/modules/logs/load-balancer-access-logs
  source = "git::git@github.com:gruntwork-io/terraform-aws-monitoring.git//modules/logs/load-balancer-access-logs?ref=v0.32.1"
  ```

Appreciate it!

**Comment by rhoboat** [comment]
Hi I just wanted to pipe in to say I saw this, I think it's an interesting idea. I definiltely have run into this myself while debugging. Having a quick way to go between repos that are cross-referenced would be so helpful. I'm not sure where that should go, and hope more people chime in, but it's on the brain. Thank you!

**Reply by Etiene** [reply]
Just want to add that I also like this idea. About how we should approach it, I see two possibilities:
* improve little by little: whenever we have to modify a module anyway, we can add this line when we see it 
    * pros: little effort,  repos that require more maintaince will get it first, we can start it immediately
    * cons: every maintainer in the team would have to be aware of this, some modules would take forever to get this improvement
* creating a script that reads all our codebase and adds a comment line above whenever it sees a source line
    * pros: would consistenly improve this in all our modules
    * cons: would take some allocation of work to code this, which, in my opinion, while useful, it's somewhat of a small improvement to justify it. Question for @rhoboat: would it require a management decision for someone invest time into this?

Question for @BrianMusson, do you prefer if the links go to the head of master like in your example, or do you think that including the tag of the respective version like this would be more useful? 
`# https://github.com/gruntwork-io/terraform-aws-monitoring/tree/v0.32.1/modules/logs/load-balancer-access-logs`

**Reply by gcagle3** [reply]
I also like this idea. I wanted to add that the script idea, although it would take more effort up front, would provide the necessary automation to help ensure this is done correctly over time. It could also be expanded as a check-in gate during the PR tests.

**Reply by BrianMusson** [reply]
I think it would the most sense to tie the link in with the tag from the source to keep things consistent.

**Reply by rhoboat** [reply]
Update: we're discussing this more internally and trying to solve the overall problem of indirection in our docs.

**Comment by rhoboat** [comment]
I will create [an issue](https://github.com/gruntwork-io/terraform-aws-service-catalog/issues/1654) in the service catalog so that we can track getting this interesting feature done.
